### ISSUE TITLE
Remove unnecessary concurrency on Project Detail page

### ISSUE THREAD

**Issue by mikerkelly** [opensafely-core/job-server#4634]
The `ProjectDetailView` uses the `concurrent` Python module to parallelise Github API requests through `github.get_repo_is_private()`. This is unnecessary and over-complicated and makes reasoning about and maintaining the code more complicated than necessary (ref #4591). It's also inefficient to open a separate connection for each item as the connection and API endpoint overheads are higher than a bulk request.

The more typical approach would be for the endpoint to provide a method to do a bulk request with a single API access (possibly paginated/chunked if needed due to a max size limit). If we look in the `github` module (our thin-wrapper around `requests` and the Github API), we can see that functions like `get_repos_with_branches` and `get_repos_with_dates` follow this approach with GraphQL API accesses. We can write a new `get_repos_with_privacy` or whatever function and eliminate this idiosyncratic use of concurrency.

Generally, we should avoid using concurrency unless we need it to avoid over-complication. This is the only use of the `concurrent` module across the entire repo. It looks like it was done for expediency or as an experiment. The other two clients of `github.get_repo_is_private()` (`WorkspaceDetail` and `SignOffRepo` views) use this function for a single request, which is reasonable. 

We might consider refactoring the commonality of the GraphQL code in `github.py` alongside doing this.

**Comment by bloodearnest** [comment]
> It's also inefficient to open a separate connection for each item as the connection and API endpoint overheads are higher than a bulk request. https://github.com/opensafely-core/job-server/issues/3393 noted this code is inefficient.

I don't think this section is quite right, as the client uses a persistent requests `Session` object to maintain a per process global connection pool: https://github.com/opensafely-core/job-server/blob/main/jobserver/github.py#L19

So TLS connections to api.github.com should be long lived and reused.  It may be that the default pool size needs increasing, or other config tweaks, to handle projects with lots of workspaces, perhaps.

Unrelated to the larger question of prefering to use bulk API endpoints, but possibly a useful detail.J 
This is a pattern we use in [other service clients](https://github.com/opensafely-core/job-server/blob/089628b2921a81f339123008a4c00ff1233ce430/jobserver/opencodelists.py#L5) too, so I think worth calling out.
